apps/frontend/src/modules/auth/slices/auth.slice.ts

```diff
@@ -18,15 +18,29 @@ const initialState: State = {
 
 const { actions, name, reducer } = createSlice({
 	extraReducers(builder) {
-		builder.addCase(signUp.pending, (state) => {
-			state.dataStatus = DataStatus.PENDING;
-		});
-		builder.addCase(signUp.fulfilled, (state) => {
-			state.dataStatus = DataStatus.FULFILLED;
+		builder.addCase(getAuthenticatedUser.rejected, (state) => {
+			state.authenticatedUser = null;
+			state.dataStatus = DataStatus.REJECTED;
 		});
+
 		builder.addCase(signUp.rejected, (state) => {
 			state.dataStatus = DataStatus.REJECTED;
 		});
+
+		builder.addMatcher(
+			isAnyOf(getAuthenticatedUser.pending, signUp.pending),
+			(state) => {
+				state.dataStatus = DataStatus.PENDING;
+			},
+		);
+
+		builder.addMatcher(
```

Comment:
Let's avoid using matchers here and define these cases separately, this is more readable and convenient to work in case we need to change something (for example, use separate status properties for each request). Matchers are good, but only for really common logic where you don't need to consider what each of the individual actions do.

Reply:
yes, do the same for fullfilled too please